machine-config/harvester.vue

```diff
@@ -0,0 +1,446 @@
+      }
+
+      if (isEmpty(this.value.diskSize)) {
+        this.value.diskSize = '40';
+      }
+    } catch (e) {
+      this.errors = exceptionToErrorsArray(e);
+    }
+  },
+
+  data() {
+    return {
+      credential:         null,
+      isImportCluster:    false,
+      cloudConfig:        '',
+      networkData:        '',
+      imageOptions:       [],
+      namespaceOptions:   [],
+      networkOptions:     [],
+      userDataOptions:    [],
+      networkDataOptions: [],
+      cpuCount:           ''
+    };
+  },
+
+  computed: { ...mapGetters({ t: 'i18n/t' }) },
+
+  watch: {
+    'credentialId'() {
+      this.imageOptions = [];
+      this.networkOptions = [];
+      this.namespaceOptions = [];
+      this.value.imageName = '';
+      this.value.networkName = '';
+      this.value.vmNamespace = '';
+
+      this.$fetch();
+    },
+
+    networkData(neu) {
+      this.value.networkData = neu;
+    },
+
+    cloudConfig(neu) {
+      this.value.cloudConfig = neu;
+    },
+  },
+
+  methods: {
+    stringify,
+
+    test() {
+      const errors = [];
+
+      if (!this.value.cpuCount || this.value.cpuCount === 'null') {
+        const message = this.validatorRequiredField(this.t('cluster.credential.harvester.cpu'));
+
+        errors.push(message);
+      }
+
+      if (!this.value.vmNamespace || this.value.vmNamespace === 'null') {
+        const message = this.validatorRequiredField(this.t('cluster.credential.harvester.namespace'));
+
+        errors.push(message);
+      }
+
+      if (!this.value.memorySize || this.value.memorySize === 'null') {
+        const message = this.validatorRequiredField(this.t('cluster.credential.harvester.memory'));
+
+        errors.push(message);
+      }
+
+      if (!this.value.diskSize || this.value.diskSize === 'null') {
+        const message = this.validatorRequiredField(this.t('cluster.credential.harvester.disk'));
+
+        errors.push(message);
+      }
+
+      if (!this.value.imageName || this.value.imageName === 'null') {
+        const message = this.validatorRequiredField(this.t('cluster.credential.harvester.image'));
+
+        errors.push(message);
+      }
+
+      if (!this.value.sshUser || this.value.sshUser === 'null') {
+        const message = this.validatorRequiredField(this.t('cluster.credential.harvester.sshUser'));
+
+        errors.push(message);
+      }
+
+      if (!this.value.networkName || this.value.networkName === 'null') {
+        const message = this.validatorRequiredField(this.t('cluster.credential.harvester.network'));
+
+        errors.push(message);
+      }
+
+      return { errors };
+    },
+
+    validatorRequiredField(key) {
+      return this.t('validation.required', { key });
+    },
+
+    valuesChanged(value, type) {
+      try {
+        this.value[type] = value;
+      } catch (e) {
+      }
+    },
+  }
+};
+</script>
+
+<template>
+  <Loading v-if="$fetchState.pending" :delayed="true" />
+  <div v-else-if="errors.length">
```

Comment:
This made sense when used only for loading errors, but don't your validation errors go in here too?  If you do something that hits one you'll be unable to correct it because the form disappears.

Reply:
hi, Vincent
      I think the error generated in fetch will be shown here, but the error generated in validation (test) function will be returned to the parent component,
   
 1.  If the error comes from fetch, such as API request error or cluster error, we should not display the form information，because we cannot create a nodedriver at this time
    
2.  If it is an error caused by running test verification when you click save, the form will not disappear, and the error message will be displayed at the bottom